bindings/java/src/main/native/context.cc

```diff
@@ -1,72 +1,69 @@
 /*
- * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+ * Copyright (C) Mellanox Technologies Ltd. 2019. ALL RIGHTS RESERVED.
  * See file LICENSE for terms.
  */
-#include "context.h"
-#include "request_util.h"
 
-ucs_status_t context::ref_context() {
-    ucs_status_t status = UCS_OK;
-    {   // Lock before checking context and updating reference counter
-        std::lock_guard<std::mutex> lk(ref_lock);
-        if (ucp_context == nullptr) {
-            status = create_context();
-        }
+#include "jucx_common_def.h"
+#include "org_ucx_jucx_ucp_UcpContext.h"
 
-        if (status == UCS_OK) {
-            ++ref_count;
-        }
-    }   // Unlock
+#include <ucp/api/ucp.h>
 
-    return status;
-}
-
-void context::deref_context() {
-    std::lock_guard<std::mutex> lk(ref_lock);
-    if (--ref_count == 0) { // All workers released
-        release_context();
-    }
-}
-
-context::~context() {
-    if (ucp_context) {
-        ucp_cleanup(ucp_context);
-    }
-}
-
-ucp_context_h context::get_ucp_context() {
-    return ucp_context;
-}
 
-ucs_status_t context::create_context() {
+/**
+ * Bridge method for creating ucp_context from java
+ */
+JNIEXPORT jlong JNICALL
+Java_org_ucx_jucx_ucp_UcpContext_createContextNative(JNIEnv *env, jclass cls,
+                                                     jobject jucx_ctx_params)
+{
     ucp_params_t ucp_params = { 0 };
     ucp_config_t *config;
     ucs_status_t status;
+    ucp_context_h ucp_context;
+    jfieldID field;
+    jclass jucx_param_class = env->GetObjectClass(jucx_ctx_params);
 
     status = ucp_config_read(nullptr, nullptr, &config);
     if (status != UCS_OK) {
-        return status;
+        JNU_ThrowExceptionByStatus(env, status);
     }
 
-    uint64_t features   =   UCP_FEATURE_TAG;
-    uint64_t field_mask =   UCP_PARAM_FIELD_FEATURES        |
-                            UCP_PARAM_FIELD_REQUEST_INIT    |
-                            UCP_PARAM_FIELD_REQUEST_SIZE;
+    field = env->GetFieldID(jucx_param_class, "fieldMask", "J");
```

Comment:
we can also remove the call to ucp_config_read() and pass NULL config to ucp_init